scaleway/resource_registry_namespace_beta.go

```diff
@@ -0,0 +1,139 @@
+package scaleway
+
+import (
+	"github.com/hashicorp/terraform-plugin-sdk/helper/schema"
+	"github.com/scaleway/scaleway-sdk-go/api/registry/v1"
+	"github.com/scaleway/scaleway-sdk-go/scw"
+)
+
+func resourceScalewayRegistryNamespaceBeta() *schema.Resource {
+	return &schema.Resource{
+		Create: resourceScalewayRegistryNamespaceBetaCreate,
+		Read:   resourceScalewayRegistryNamespaceBetaRead,
+		Update: resourceScalewayRegistryNamespaceBetaUpdate,
+		Delete: resourceScalewayRegistryNamespaceBetaDelete,
+		Importer: &schema.ResourceImporter{
+			State: schema.ImportStatePassthrough,
+		},
+		SchemaVersion: 0,
+		Schema: map[string]*schema.Schema{
+			"name": {
+				Type:        schema.TypeString,
+				Required:    true,
+				ForceNew:    true,
+				Description: "The name of the container registry namespace",
+			},
+			"description": {
+				Type:        schema.TypeString,
+				Optional:    true,
+				Description: "The description of the container registry namespace",
+			},
+			"is_public": {
+				Type:        schema.TypeBool,
+				Optional:    true,
+				Description: "Define the default visibity policy",
+			},
+			"endpoint": {
+				Type:        schema.TypeString,
+				Computed:    true,
+				Description: "The endpoint reachable by docker",
+			},
+			"region":          regionSchema(),
+			"organization_id": organizationIDSchema(),
+		},
+	}
+}
+
+func resourceScalewayRegistryNamespaceBetaCreate(d *schema.ResourceData, m interface{}) error {
+	api, region, err := registryNamespaceWithRegion(d, m)
+	if err != nil {
+		return err
+	}
+
+	name, ok := d.GetOk("name")
```

Comment:
As the name is a unique identifier, we don't want to generate it and let the user explicitly specify it.

> Note that a namespace name is unique in a region. Thus, if another client already has created "test", you can't have it as a namespace.

You have marked the `name` attribute as required, it's fine, now you also have to remove the:
```go
if !ok {
	name = getRandomName("cr")
}
```